data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/AwsCloudMapPeerListProviderTest.java

```diff
@@ -0,0 +1,386 @@
+                    .willReturn(100L);
+
+        }
+
+        @Test
+        void getPeerList_returns_value_after_several_failed_attempts() {
+
+            final AwsCloudMapPeerListProvider objectUnderTest = createObjectUnderTest();
+
+            waitUntilDiscoverInstancesCalledAtLeastOnce();
+
+            final List<String> expectedEmpty = objectUnderTest.getPeerList();
+
+            assertThat(expectedEmpty, notNullValue());
+            assertThat(expectedEmpty.size(), equalTo(0));
+
+            waitUntilPeerListPopulated(objectUnderTest);
+
+            final List<String> expectedPopulated = objectUnderTest.getPeerList();
+
+            assertThat(expectedPopulated, notNullValue());
+            assertThat(expectedPopulated.size(), equalTo(knownIpPeers.size()));
+
+            assertThat(new HashSet<>(expectedPopulated), equalTo(new HashSet<>(knownIpPeers)));
+
+            final InOrder inOrder = inOrder(backoff);
+            then(backoff)
+                    .should(inOrder)
+                    .nextDelayMillis(1);
+            then(backoff)
+                    .should(inOrder)
+                    .nextDelayMillis(2);
+            then(backoff)
+                    .shouldHaveNoMoreInteractions();
+        }
+
+        @Test
+        void listener_gets_list_after_several_failed_attempts() {
+
+            final List<Endpoint> listenerEndpoints = new ArrayList<>();
+
+            final AwsCloudMapPeerListProvider objectUnderTest = createObjectUnderTest();
+
+            objectUnderTest.addListener(listenerEndpoints::addAll);
+
+            waitUntilDiscoverInstancesCalledAtLeastOnce();
+
+            assertThat(listenerEndpoints.size(), equalTo(0));
+
+            waitUntilPeerListPopulated(objectUnderTest);
+
+            assertThat(listenerEndpoints.size(), equalTo(knownIpPeers.size()));
+
+            final Set<String> observedIps = listenerEndpoints.stream()
+                    .map(Endpoint::ipAddr)
+                    .collect(Collectors.toSet());
+
+            assertThat(observedIps, equalTo(new HashSet<>(knownIpPeers)));
+
+            final InOrder inOrder = inOrder(backoff);
+            then(backoff)
+                    .should(inOrder)
+                    .nextDelayMillis(1);
+            then(backoff)
+                    .should(inOrder)
+                    .nextDelayMillis(2);
+            then(backoff)
+                    .shouldHaveNoMoreInteractions();
+        }
+    }
+
+    private void waitUntilDiscoverInstancesCalledAtLeastOnce() {
+        waitUntilDiscoverInstancesCalledAtLeast(1);
+    }
+
+    /**
+     * Waits for DiscoverInstances to be called at least a specified number
+     * of times. This method intentionally does not inspect the request.
+     *
+     * @param timesCalled The number of times to wait for it to be called.
+     */
+    private void waitUntilDiscoverInstancesCalledAtLeast(final int timesCalled) {
+        final long waitTimeMillis = (long) (timesCalled * 1000 * 1.2);
```

Comment:
Nit: 1000 and 1.2 appear magic here. Do we have any constant in the provider to replace them?